From 6e5e3b3046c16a9ef840b411a6baebf3862f2dc9 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 5 Oct 2021 18:59:20 -0600
Subject: [PATCH] Update setup.py

---
 setup.py | 30 ++++++++++++++++++++++++------
 1 file changed, 24 insertions(+), 6 deletions(-)

diff --git a/setup.py b/setup.py
index b0290f3..a0cdd83 100644
--- a/setup.py
+++ b/setup.py
@@ -16,29 +16,47 @@ def run(self):
         if '__init__.py' in filenames:
             packages.append(dirname.replace('/', '.'))
 
+library_lst = []
+for r, d, fl in os.walk('tsase'):
+    for f in fl:
+        if f.endswith(".so"):
+            library_lst.append(os.path.join(r, f))
+
 package_dir = {'tsase': 'tsase'}
 
-scripts = ['bin/xyz']
+scripts = [
+    "bin/dump2xdat",
+    "bin/kmc",
+    "bin/lmp2con",
+    "bin/lmp2pos",
+    "bin/mobfil",
+    "bin/neighbors",
+    "bin/oldexpectra",
+    "bin/pdf-make.py",
+    "bin/soc2pos",
+    "bin/splitxdat",
+    "bin/temgui",
+    "bin/tsase",
+    "bin/water_solvate",
+    "bin/water_solvate_z",
+    "bin/xyz",
+]
 
 package_data = {'tsase': ['xyz/xyz.glade',
                           'xyz/xyz.help',
                           'xyz/*.png',
-                          'calculators/al/al_.so',
                           'calculators/cuo/ffield.comb',
                           'calculators/cuo/ffield.comb3',
                           'calculators/cuo/in.lammps',
-                          'calculators/lepspho/lepspho_.so',
                           'calculators/lisi/LiSi.meam',
                           'calculators/lisi/in.lammps',
                           'calculators/lisi/library.meam',
-                          'calculators/lj/lj_.so',
                           'calculators/mo/Mo.set',
                           'calculators/mo/in.lammps',
-                          'calculators/morse/morse_.so',
                           'calculators/si/Si.meam',
                           'calculators/si/library.meam',
                           'calculators/w/W.set',
-                          'calculators/w/in.lammps']}
+                          'calculators/w/in.lammps'] + library_lst}
 
 setup(name='tsase',
       version='1.0',
