From a1f9607b2bdda378d9958487282a051524fa819a Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 5 Oct 2021 19:37:46 -0600
Subject: [PATCH] Update setup.py

---
 setup.py | 95 +++++++++++++++++++++++++++++++++++------------
 1 file changed, 73 insertions(+), 22 deletions(-)

diff --git a/setup.py b/setup.py
index b0290f3..dc23cec 100644
--- a/setup.py
+++ b/setup.py
@@ -1,24 +1,74 @@
 #!/usr/bin/env python
-from distutils.core import setup
-from distutils.command.install import install as DistutilsInstall
+from numpy.distutils.core import setup, Extension
 import subprocess
 from sys import exit
-
 import os
 
-class CustomInstall(DistutilsInstall):
-        def run(self):
-            subprocess.check_call('cd tsase; make', shell=True)
-            DistutilsInstall.run(self)
 
 packages = []
 for dirname, dirnames, filenames in os.walk('tsase'):
-        if '__init__.py' in filenames:
-            packages.append(dirname.replace('/', '.'))
+    if '__init__.py' in filenames:
+        packages.append(dirname.replace('/', '.'))
+
+
+al_ext = Extension(
+    name='tsase.calculators.al.al_',
+    sources=[
+        'tsase/calculators/al/code/sumembforce.f',
+        'tsase/calculators/al/code/potinit.f',
+        'tsase/calculators/al/code/gagafeDblexp.f',
+        'tsase/calculators/al/code/forces.f',
+        'tsase/calculators/al/code/fofrhoDblexp.f',
+        'tsase/calculators/al/code/embedenergy.f',
+        'tsase/calculators/al/code/dfrhoDblexp.f',
+    ],
+    include_dirs=['tsase/calculators/al/code/commonblks']
+)
+
+
+lepspho_ext = Extension(
+    name='tsase.calculators.lepspho.lepspho_',
+    sources=[
+        'tsase/calculators/lepspho/code/lepspho.f',
+    ],
+)
+
+
+lj_ext = Extension(
+    name='tsase.calculators.lj.lj_',
+    sources=[
+        'tsase/calculators/lj/code/lj.f',
+    ],
+)
+
+
+morse_ext = Extension(
+    name='tsase.calculators.morse.morse_',
+    sources=[
+        'tsase/calculators/morse/code/morse.f',
+    ],
+)
+
 
 package_dir = {'tsase': 'tsase'}
 
-scripts = ['bin/xyz']
+scripts = [
+    "bin/dump2xdat",
+    "bin/kmc",
+    "bin/lmp2con",
+    "bin/lmp2pos",
+    "bin/mobfil",
+    "bin/neighbors",
+    "bin/oldexpectra",
+    "bin/pdf-make.py",
+    "bin/soc2pos",
+    "bin/splitxdat",
+    "bin/temgui",
+    "bin/tsase",
+    "bin/water_solvate",
+    "bin/water_solvate_z",
+    "bin/xyz",
+]
 
 package_data = {'tsase': ['xyz/xyz.glade',
                           'xyz/xyz.help',
@@ -40,15 +98,16 @@ def run(self):
                           'calculators/w/W.set',
                           'calculators/w/in.lammps']}
 
-setup(name='tsase',
-      version='1.0',
-      description='Library based upon ASE for transition state theory calculations.',
-      author='Henkelman Research Group',
-      author_email='henkelman@utexas.edu',
-      url='http://www.henkelmanlab.org',
-#      packages=['tsase'],
-      scripts=scripts,
-      packages=packages,
-      package_data=package_data,
-      cmdclass={'install' : CustomInstall}
-     )
+setup(
+    name='tsase',
+    version='1.0',
+    description='Library based upon ASE for transition state theory calculations.',
+    author='Henkelman Research Group',
+    author_email='henkelman@utexas.edu',
+    url='http://www.henkelmanlab.org',
+    #      packages=['tsase'],
+    scripts=scripts,
+    packages=packages,
+    package_data=package_data,
+    ext_modules=[al_ext, lepspho_ext, lj_ext, morse_ext]
+)
