From ddf65edc69cced25098bed6e839015428ebb54e5 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 4 Sep 2025 14:25:16 +0200
Subject: [PATCH] Update setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 6681b49..1e05a54 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-from numpy.distutils.core import setup, Extension
+from setuptools import setup, Extension
 import subprocess
 from sys import exit
 import os
